README.md

```diff
@@ -7,367 +7,600 @@
+
+When using the CLI, the default formatter is `progress-bar`. This can only be overriden using the `--format [format]` flag.
+
+#### `run: boolean` (since: `v3.8.0`)
 
-Check for duplicates in test messages. Can be disabled with:
+Whether or not to run a test case. Defaults to `true`.
 
-- passing `--no-check-duplicates` command line flag;
-- setting `SUPERTAPE_CHECK_DUPLICATES=0` env variable;
+#### `checkDuplicates: boolean` (since: `v5.6.0`)
 
-#### ❌ Example of incorrect code
+Whether or not to check test messages for duplicates. 📼 **Supertape** expects each message to be unique. Defaults to `true` (was `false` until `v6.0.0`).
+
+#### `checkAssertionsCount: boolean` (since: `v6.8.0`)
+
+Whether or not to check the number of assertions per test case. 📼 **Supertape** expects each test to have only one assertion. Defaults to `true`.
+
+#### `checkScopes: boolean` (since: `v6.7.0`)
+
+Whether or not to check that test messages are scoped (i.e. in the form `'scope: message'`). 📼 **Supertape** expects each test case to be scoped.
+
+## Validation
+
+📼 **Supertape** has a number of built-in validation checks to ensure that your tests are of the highest possible quality. A test will fail if it does not pass these checks. Certain checks can be disabled. (See: [Configuration](#usage-and-configuration))
+
+### Single Call to `t.end()`
+
+Each test must call the [`t.end()`](#tend-since-v310) operator to denote that it has finished. Only one call to `t.end()` is allowed, and no assertions are allowed after it.
+
+This check cannot be disabled. In addition, 🐊 **Putout** has a rule 🐊 [`remove-useless-t-end`][PutoutTEnd] for ensuring it is only called once. (See: [Putout](#-putout))
+
+[PutoutTEnd]: https://github.com/coderaiser/putout/blob/master/packages/plugin-tape/README.md#remove-useless-t-end
+
+#### **❌ <u>Example of Incorrect Code</u>:**
 
 ```js
 test('hello: world', (t) => {
-    t.equal(1, 1);
     t.end();
+    t.end();
+    t.pass('hello, world!');
 });
 ```
 
+#### **✅ <u>Example of Correct Code</u>:**
+
 ```js
 test('hello: world', (t) => {
-    t.equal(2, 1);
+    t.pass('hello, world!');
     t.end();
 });
 ```
 
-### Check scopes
+### Duplicate Test Messages
 
-Check that test message are divided on groups by colons. Can be disabled with:
+📼 **Supertape** expects all test messages to be unique. This check can be disabled with:
 
-- passing `--no-check-scopes` command line flag;
-- setting `SUPERTAPE_CHECK_SCOPES=0` env variable;
+- the [`--no-check-duplicates`](#--no-check-duplicates) command line flag
+- the [`SUPERTAPE_CHECK_DUPLICATES`](#supertape_check_duplicates) environment variable (set to `0`)
+- the [`checkDuplicates`](#checkduplicates-boolean-since-v560) test option (set to `true`)
 
-#### ❌ Example of incorrect code
+#### **❌ <u>Example of Incorrect Code</u>:**
```

Comment:
why `<u>` and capitalized `Incorrect Code`?

Reply:
The capitalization was for consistency with other headings. I tried to underline it to differentiate it from `####` sized text that isn't a section heading, but turns out GitHub Flavored Markdown uses `<ins></ins>` <ins>for underlining</ins> instead of `<u></u>`.

Reply:
https://github.com/tommy-mitchell/supertape/commit/58f35bb0c000be757b39681a69a622813fdfacec uses `<ins>`

Reply:
I don’t think we need underlining.